src/plugins/vis_builder/public/application/utils/get_top_nav_config.tsx

```diff
@@ -152,6 +112,61 @@ export const getTopNavConfig = (
   return topNavConfig;
 };
 
+export const getTopNavConfig = (
+  {
+    visualizationIdFromUrl,
+    savedVisBuilderVis,
+    saveDisabledReason,
+    dispatch,
+    originatingApp,
+  }: TopNavConfigParams,
+  services: VisBuilderServices,
+  navActions: VisBuilderNavActionMap
+) => {
+  const topNavMenu = [
+    {
+      tooltip: !!saveDisabledReason
+        ? saveDisabledReason
+        : savedVisBuilderVis?.id && originatingApp
+        ? i18n.translate('visBuilder.topNavMenu.saveVisualizationAsButtonLabel', {
+            defaultMessage: 'save as',
+          })
+        : i18n.translate('visBuilder.topNavMenu.saveVisualizationButtonLabel', {
+            defaultMessage: 'save',
+          }),
+      ariaLabel: i18n.translate('visBuilder.topNavMenu.saveVisualizationAsButtonLabel', {
```

Comment:
Check with UX but I feel when disabled, the aria-label might want to show the reason too.

Reply:
Sure will check with UX Regarding this.